controllers/provider-aws/pkg/webhook/controlplane/ensurer.go

```diff
@@ -38,30 +39,45 @@ func NewEnsurer(logger logr.Logger) genericmutator.Ensurer {
 
 type ensurer struct {
 	genericmutator.NoopEnsurer
+	client client.Client
 	logger logr.Logger
 }
 
+// InjectClient injects the given client into the ensurer.
+func (e *ensurer) InjectClient(client client.Client) error {
+	e.client = client
+	return nil
+}
+
 // EnsureKubeAPIServerDeployment ensures that the kube-apiserver deployment conforms to the provider requirements.
 func (e *ensurer) EnsureKubeAPIServerDeployment(ctx context.Context, dep *appsv1.Deployment) error {
-	ps := &dep.Spec.Template.Spec
+	template := &dep.Spec.Template
+	ps := &template.Spec
 	if c := controlplane.ContainerWithName(ps.Containers, "kube-apiserver"); c != nil {
 		ensureKubeAPIServerCommandLineArgs(c)
 		ensureEnvVars(c)
 		ensureVolumeMounts(c)
 	}
 	ensureVolumes(ps)
+	if err := e.ensureChecksumAnnotations(ctx, template, dep.Namespace); err != nil {
```

Comment:
```suggestion
	return e.ensureChecksumAnnotations(ctx, template, dep.Namespace)
```

(replicate for other providers)

Reply:
Done